core/src/main/java/org/mule/endpoint/AbstractEndpoint.java

```diff
@@ -508,6 +508,12 @@ public void dispose()
         // this.messageProcessors.clear();
         // Don't clear this, since it changes the hash code, which can foul up shutdown processing
         // when objects have been keyed by endpoint, e.g. dispatchers
+
```

Comment:
The comments above are related to the commented clear() on the messageProcessors field, they are previous to this change and not related to it. We could move and separate the comments in order to make them clearer.